### ISSUE TITLE
knockingParticipant event not working

### ISSUE THREAD

**Issue by Harsh246** [jitsi/docker-jitsi-meet#1446]
Hi,
Currently, we are using the stable-7287-2 version and out of nowhere knockingParticipant event stops working. 

Can you please provide some input or help me with it?

Thanks in advance

**Comment by saghul** [comment]
So was it working before? Do you see any errors in the Prosody container?

**Comment by Harsh246** [comment]
I see no errors there. Yes It's working in one meeting but not in other.

**Comment by saghul** [comment]
Hum. Are you able to reproduce this with the latest stable images?

**Comment by Harsh246** [comment]
But how is this even possible? Does it have anything to do with maxListeners i.e 10?

**Comment by saghul** [comment]
I don't know.

**Comment by Harsh246** [comment]
> Hum. Are you able to reproduce this with the latest stable images?

I was able to reproduce with stable-8044 image.

**Comment by saghul** [comment]
Does the event appear at all on that version?

**Comment by Harsh246** [comment]
yes, it is. I'm even trying to log the registered event names multiple times using participantJoined event i.e:
`Registered Event Names: knockingParticipant,participantRoleChanged,videoConferenceJoined,participantJoined,participantLeft,incomingMessage,contentSharingParticipantsChanged,errorOccurred,videoConferenceLeft,screenSharingStatusChanged`

but knockingParticipant is not firing at unusual times. Once it stops no matter how many requests come, it is not firing.

**Comment by Harsh246** [comment]
hey @saghul,
Is there any way to capture meeting request time i.e when the participant clicks on Ask to join button either on moderator or participant's end other than knocking participant.

Would really appreciate your help.

**Comment by Harsh246** [comment]
Hey @saghul ,
We were able to reproduce the same. Whenever the moderator has the participant pane open and a new participant asks to join the meeting, the knockingParticipant event is not fired.

**Comment by Harsh246** [comment]
> Hey @saghul , We were able to reproduce the same. Whenever the moderator has the participant pane open and a new participant asks to join the meeting, the knockingParticipant event does not get fired.

++

**Comment by Harsh246** [comment]
> Hey @saghul , We were able to reproduce the same. Whenever the moderator has the participant pane open and a new participant asks to join the meeting, the knockingParticipant event does not get fired.

@saghul +++

**Comment by saghul** [comment]
I found where the problem is. We are currently refactoring some code around there, so we'll take this into consideration while making the change.

@Calinteodor let's sync on this.
